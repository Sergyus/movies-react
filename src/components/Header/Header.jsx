import React from 'react';
import styles from './Header.scss';
import SearchFormContainer from './components/SearchForm';

const Header = () => (
  <header className={styles.header}>
    <div className={styles.container}>
      <h1 className={styles.logo}>Films</h1>
      <SearchFormContainer />
    </div>
  </header>
);

export default Header;
