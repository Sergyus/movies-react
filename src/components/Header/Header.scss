@import "../../styles/variables.scss";

.header {
  position: fixed;
  z-index: 2;
  width: 100vw;
  background: linear-gradient(to top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.9));
}

.container {
  margin: 30px 5%;
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  flex-wrap: wrap;
}

.logo {
  text-transform: uppercase;
  font-size: 1.7rem;
  color: #fff;
  letter-spacing: 3px;
  font-weight: normal;
}

.input {
  border: #fff 2px solid;
  color: #fff;
  height: 35px;
  width: 270px;
  background: rgba(#000, 0);
  border-radius: 5px;
  opacity: 1;
  padding: 10px;
  font-size: 0.9rem;
  outline: none;

  &::placeholder {
    color: #fff;
    opacity: 1;
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }
}

.searchForm {
  position: relative;
}

.searchBtn {
  content: "";
  position: absolute;
  bottom: 7px;
  right: 15px;
  outline: none;
  width: 20px;
  height: 20px;
  background-color: transparent;
  background-image: url("../../assets/images/sprites.png");
  background-position: -110px -170px;
  cursor: pointer;
  border: none;
}

@media (max-width: $mobile-screen-size) {
  .container {
    justify-content: center;
    align-items: center;
    margin-top: 20px;
    margin-bottom: 15px;
  }

  .logo {
    font-size: 1.5rem;
    margin-bottom: 10px;
    margin-right: 20px;
  }
}
