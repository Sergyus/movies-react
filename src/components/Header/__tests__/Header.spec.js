import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '..';

describe('Header component renders correctly', () => {
  it('Header renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Header />);
    expect(result).toMatchSnapshot();
  });
});
