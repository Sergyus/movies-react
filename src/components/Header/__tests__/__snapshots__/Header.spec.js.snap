// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      Films
    </h1>
    <Memo(ConnectFunction) />
  </div>
</header>
`;
